### DISCUSSION TITLE
Switching between multiple power sources

### DISCUSSION THREAD

**Discussion by jessevdk** [PiSupply/PiJuice#964] [Q&A]
I have a pi zero with corresponding pijuice. I'm currently using it with a battery and solar panel setup. Sometimes I do not have enough solar to keep the rpi running. I'd like to have a secondary source of power that switches on to charge the battery once it drops below a certain charge. Is there an easy way to do this?

**Comment by tvoverbeek** [comment]
Charging the battery only works via the USB power input on the PiJuice Zero or its parallel J4 connector.
I assume you have connected the solar panel to the PiJuice micro USB.
With external switching logic you could switch the power connection to the micro USB off and switch an external power supply (your secondary source) connected to J4 on.
Note: there are 2 GPIOs from the MCU on the P3 expansion connector which you could use to command the switches.
Of course you could also use some ot the Pi GPIOs for the switching.
